### PR TITLE
vk_shader: Implement ImageLoad

### PR DESCRIPTION
**Pull request by namkazt**
Used by Kirby Star Allies.
Issue: #3289 

Thanks @ReinUsesLisp for helping me out.

### CODE REVIEW

**Comment by theboy181** [comment]
any other games?

**Comment by namkazt** [comment]
> any other games?

I'm not sure. Only test with kirby

**Comment by lioncash** [comment]
Commit message itself should probably indicate that Kirby Star Allies uses it, so that information is stored in the repository

**Comment by namkazt** [comment]
> Commit message itself should probably indicate that Kirby Star Allies uses it, so that information is stored in the repository

@lioncash anyway how can i change the commit message in Github.

**Comment by lioncash** [comment]
Use `git commit --amend` and then alter the message, then force push.

**Comment by namkazt** [comment]
> Use `git commit --amend` and then alter the message, then force push.

ok so it must be use on local. Thanks.

**Comment by ReinUsesLisp** [comment]
@theboy181 Xenoblade 2 also uses SULD, but it uses a variant that's not currently implemented (SULD.D instead of SULD.P).

**Comment by namkazt** [comment]
> Aside from that, we also need to stub this call (by returning `Immediate(0.0f)`) when the device is not capable of loading images without an explicit format. In other words, if the Vulkan feature `shaderStorageImageReadWithoutFormat` is false. Feel free to ask me on Discord or here if you need help loading this in `vk_device`.

Added check for `shaderStorageImageReadWithoutFormat`

File: src/video_core/renderer_vulkan/vk_shader_decompiler.cpp [hunk 1]
```diff
@@ -274,6 +274,7 @@ class SPIRVDecompiler final : public Sirit::Module {
         AddCapability(spv::Capability::ImageBuffer);
         AddCapability(spv::Capability::ImageGatherExtended);
         AddCapability(spv::Capability::SampledBuffer);
+        AddCapability(spv::Capability::StorageImageReadWithoutFormat);
```
Comment (ReinUsesLisp) [inline src/video_core/renderer_vulkan/vk_shader_decompiler.cpp]: Only add this one when the extension is available.

File: src/video_core/renderer_vulkan/vk_device.cpp [hunk 2]
```diff
@@ -530,6 +534,7 @@ std::unordered_map<vk::Format, vk::FormatProperties> VKDevice::GetFormatProperti
                                         vk::Format::eBc6HUfloatBlock,
                                         vk::Format::eBc6HSfloatBlock,
                                         vk::Format::eBc1RgbaSrgbBlock,
+                                        vk::Format::eBc2SrgbBlock,
```
Comment (ReinUsesLisp) [inline src/video_core/renderer_vulkan/vk_device.cpp]: Is this a left over from another branch?

**Comment by ReinUsesLisp** [comment]
Just one nit, other than that, it looks good to me.

File: src/video_core/renderer_vulkan/vk_device.cpp (hunk 2 above)
Comment (namkazt) [reply src/video_core/renderer_vulkan/vk_device.cpp]: oh right

**Pull request merged**